crates/workspace/src/workspace.rs

```diff
@@ -3412,7 +3412,7 @@ impl Workspace {
             if !location.paths().is_empty() {
                 let center_group = build_serialized_pane_group(&self.center.root, cx);
                 let docks = build_serialized_docks(self, cx);
-
+                (*cx).note_recent(&location.paths()[0]);
```

Comment:
This is the most important part, and I'm not sure the current approach is right.

* Zed has a list of recent projects (cmd-alt-o by default), and that list is out of sync with the menu items people will see, since here we only add a project on the serialization.

My first launch:
<img width="584" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/c10f6e95-4a0c-40ca-9ec1-70ba4248e939">

<img width="207" alt="Screenshot 2024-03-14 at 21 45 22" src="https://github.com/zed-industries/zed/assets/2690773/8fc22447-cf8c-49d6-9bc4-2fddf40f5d6a">

It is amazing that the menu icon list is persisted between the launches, and I wonder if there' a way to make it synchronized with the recent project pane list?
People can also remove items from that recent project list, hence prefer having it removed from this context menu.

* The DB part we were discussing in the beginning of this PR, is supposed to be needed for the sync, but I do not think we need to refresh it on every serialization of the workspace: it happens after every tab is closed or other events, not related to recent projects list change, happen.

Reply:
I actually was considering a sync with the DB, but I wasn't sure how to implement it. Specifically:
1. How can I attach an event listener to the DB that runs whenever the DB updates?
2. Can I even call `note_recent` directly from the DB, since it would require a reference to `AppContext`?

Reply:
The whole point of the changes proposed is not to actually do any listening or "directly from DB" things: we know that DB changes on workspace removal (that might be a bit hard, so feel free to skip) and workspace addition (`Workspace::new_local` method is fine for our cases since all that we care about is `abs_path` and the fact that the workspace is not remote).

In either of these methods, and even in the line of the current PR, one could do
`let ids = DB.recent_workspaces_on_disk().await;`
and use the `WorkspaceLocation` data to get all available workspaces' paths out and set into the new menu.

Reply:
I just implemented your suggestions, let me know if I put anything in the wrong place!